packages/dev/test/provider/RelayClient.test.ts

```diff
@@ -744,4 +745,65 @@ contract('RelayClient', function (accounts) {
       assert.exists(relayingResult.transaction)
     })
   })
+
+  describe('_resolveConfiguration()', function () {
+    it('should prioritize client config in that order: config function argument, website-supplied config, default config', async function () {
+      sinon.stub(relayClient, '_resolveConfigurationFromServer').returns(Promise.resolve({
+        methodSuffix: 'test suffix from _resolveConfigurationFromServer'
+      }))
+      const config: Partial<GSNConfig> = {
+        methodSuffix: 'test suffix from arg'
+      }
+      let resolvedConfig = await relayClient._resolveConfiguration({
+        provider: relayClient.getUnderlyingProvider(),
+        config
+      })
+      assert.equal(resolvedConfig.methodSuffix, 'test suffix from arg')
+
+      resolvedConfig = await relayClient._resolveConfiguration({
+        provider: relayClient.getUnderlyingProvider(),
+        config: {}
+      })
+      assert.equal(resolvedConfig.methodSuffix, 'test suffix from _resolveConfigurationFromServer')
+
+      sinon.restore()
+      sinon.stub(relayClient, '_resolveConfigurationFromServer').returns(Promise.resolve({}))
+      resolvedConfig = await relayClient._resolveConfiguration({
+        provider: relayClient.getUnderlyingProvider(),
+        config: {}
+      })
+      assert.equal(resolvedConfig.methodSuffix, defaultGsnConfig.methodSuffix)
+      sinon.restore()
+    })
+    it('should not use website configuration if useGsnDocsConfig is false', async function () {
+      const spy = sinon.spy(relayClient, '_resolveConfigurationFromServer')
+      const config: Partial<GSNConfig> = {
+        useGsnDocsConfig: false
+      }
+      const resolvedConfig = await relayClient._resolveConfiguration({
+        provider: relayClient.getUnderlyingProvider(),
+        config
+      })
+      assert.equal(resolvedConfig.methodSuffix, defaultGsnConfig.methodSuffix)
+      sinon.assert.notCalled(spy)
+      sinon.restore()
+    })
+    describe('_resolveConfigurationFromServer()', function () {
+      let supportedNetworks: number[]
+      let jsonConfig: ConfigResponse
+      before('get all supported networks', async function () {
+        jsonConfig = await relayClient.dependencies.httpClient.getNetworkConfiguration()
+        supportedNetworks = Object.keys(jsonConfig.networks).map(k => parseInt(k))
+      })
+      it('should get configuration from opengsn for all supported networks', async function () {
+        for (const network of supportedNetworks) {
+          console.log(`Network ${jsonConfig.networks[network].name}`)
```

Comment:
Also, I suggest not logging big objects in tests

Reply:
This is not a big log, just the network name